admin/test/test_release.py

```diff
@@ -53,3 +60,604 @@ def test_non_integer_suffix(self):
+            s3_buckets={
+                'clusterhq-docs': {
+                    'index.html': '',
+                    'en/index.html': '',
+                    'en/latest/index.html': '',
+                },
+                'clusterhq-dev-docs': {
+                    '0.3.1/index.html': 'index-content',
+                    '0.3.1/sub/index.html': 'sub-index-content',
+                    '0.3.1/other.html': 'other-content',
+                    '0.3.0-392-gd50b558/index.html': 'bad-index',
+                    '0.3.0-392-gd50b558/sub/index.html': 'bad-sub-index',
+                    '0.3.0-392-gd50b558/other.html': 'bad-other',
+                },
+            })
+        self.publish_docs(aws, '0.3.1', '0.3.1',
+                          environment=Environments.PRODUCTION)
+        self.assertEqual(
+            aws.s3_buckets['clusterhq-docs'], {
+                'index.html': '',
+                'en/index.html': '',
+                'en/latest/index.html': '',
+                'en/0.3.1/index.html': 'index-content',
+                'en/0.3.1/sub/index.html': 'sub-index-content',
+                'en/0.3.1/other.html': 'other-content',
+            })
+
+    def test_deletes_removed_documentation(self):
+        """
+        Calling :func:`publish_docs` replaces documentation from
+        ``s3://clusterhq-staging-docs/en/<doc_version>/``.
+        with documentation from ``s3://clusterhq-dev-docs/<flocker_version/``.
+        In particular, files with changed content are updated, and removed
+        files are deleted.
+        """
+        aws = FakeAWS(
+            routing_rules={
+                'clusterhq-staging-docs': {
+                    'en/latest/': 'en/0.3.0/',
+                },
+            },
+            s3_buckets={
+                'clusterhq-staging-docs': {
+                    'index.html': '',
+                    'en/index.html': '',
+                    'en/latest/index.html': '',
+                    'en/0.3.1/index.html': 'old-index-content',
+                    'en/0.3.1/sub/index.html': 'old-sub-index-content',
+                    'en/0.3.1/other.html': 'other-content',
+                },
+                'clusterhq-dev-docs': {
+                    '0.3.0-444-gf05215b/index.html': 'index-content',
+                    '0.3.0-444-gf05215b/sub/index.html': 'sub-index-content',
+                },
+            })
+        self.publish_docs(aws, '0.3.0-444-gf05215b', '0.3.1',
+                          environment=Environments.STAGING)
+        self.assertEqual(
+            aws.s3_buckets['clusterhq-staging-docs'], {
+                'index.html': '',
+                'en/index.html': '',
+                'en/latest/index.html': '',
+                'en/0.3.1/index.html': 'index-content',
+                'en/0.3.1/sub/index.html': 'sub-index-content',
+            })
+
+    def test_updates_redirects(self):
+        """
+        Calling :func:`publish_docs` with a release or documentation version
```

Comment:
This doesn't test calling `publish_docs` with a documentation version so that shouldn't be mentioned. Considering that `flocker_version` and `doc_version` are two `version`s which can be given to `publish_docs` these test docstrings are quite confusing.